### ISSUE TITLE
Could I use this as Directive?

### ISSUE THREAD

**Issue by thearabbit** [dm4t2/vue-currency-input#1]
Now I base on `Element UI`.
Could I use this as directive
```js
<el-input v-model="value" v-currency="opts" />

--------

import CurrencyInput from 'vue-currency-input'

export default {
  directives: { currency: CurrencyInput },
  data: () => ({
    value: 1000,
    opts: {
            currency: 'USD',
           allow-negative:...,
           decimal: .....,
    }
  })
}
```

**Comment by dm4t2** [comment]
No, currently not. But a Vue directive for usage with components libs such as Element UI is on the roadmap.

**Comment by dm4t2** [comment]
I just released 1.2.0 providing the new `v-currency` directive. I tested the directive with Element and Vuetify.

Check out the example source:
https://github.com/dm4t2/vue-currency-input/blob/master/docs/.vuepress/components/DirectiveDemo.vue

**Comment by fergardi** [comment]
If anyone is interested, I've managed to refactor the given example of element-ui integration into Typescript:

```ts
<!-- https://github.com/dm4t2/vue-currency-input/issues/19#issuecomment-514995907 -->
<template>
  <el-input ref="elInput" v-currency="options" :value="formattedValue" @input="handleInput" :placeholder="placeholder"/>
</template>

<script lang="ts">
import { setValue, getValue } from 'vue-currency-input';
import { Component, Vue, Emit, Watch, Prop } from 'vue-property-decorator';
import { ElInput } from 'element-ui/types/input';

@Component({})
export default class ElCurrencyInput extends Vue {

  public formattedValue: string | number | null = null;
  @Prop() private value: any;
  @Prop() private placeholder?: string;

  @Watch('value')
  private onValueChange(value: number, oldValue: number) {
    this.setValue(value);
  }

  @Emit('input')
  private handleInput(value: string): number | null {
    this.formattedValue = value;
    return getValue((this.$refs.elInput as ElInput).$refs.input as HTMLInputElement);
  }

  private setValue(value: number) {
    setValue((this.$refs.elInput as ElInput).$refs.input as HTMLInputElement, value);
  }

  private mounted() {
    this.setValue(this.value);
  }

  get options() {
    return {
      locale: 'de-DE', // 'es-ES' does not have thousands separators in numbers of 4 digits or less, but 'de-DE' does
      currency: null, // to hide the coin,
    };
  }

}
</script>

```
